### PR TITLE
feat: add support for different actor_id sources

### LINKED ISSUE #48: Bug: Non-anonymous actor ids are null
When using the mbox or mbox hashed types, the materialized view doesn't look at the right keys to propagate the actor_id to the _parsed table. We should COALESCE those keys together here: https://github.com/openedx/tutor-contrib-oars/blob/ed5b6f3d797553a38e9cd61819bd2ca20f87f16f/tutoroars/templates/oars/jobs/init/oars_init_schemas_tables_users.sh#L59

### PR DESCRIPTION
**Pull request by Ian2012**
## Description

This PR allows to change the `XAPI_AGENT_IFI_TYPE` setting without breaking the `actor_id` column

- Fixes #48

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: tutoroars/templates/oars/jobs/init/oars_init_schemas_tables_users.sh [hunk 1]
```diff
@@ -56,7 +56,12 @@ CREATE MATERIALIZED VIEW IF NOT EXISTS {{ OARS_XAPI_DATABASE }}.{{ OARS_XAPI_TRA
     SELECT
     event_id as event_id,
     JSON_VALUE(event_str, '$.verb.id') as verb_id,
-    JSON_VALUE(event_str, '$.actor.account.name') as actor_id,
+    COALESCE(
+        NULLIF(JSON_VALUE(event_str, '$.actor.account.name'), ''),
+        NULLIF(JSON_VALUE(event_str, '$.actor.mbox'), ''),
+        NULLIF(JSON_VALUE(event_str, '$.actor.mbox_sha1sum'), ''),
+        'none'
```
Comment (bmtcril) [inline tutoroars/templates/oars/jobs/init/oars_init_schemas_tables_users.sh:63]: I think we just want it to be an empty string if nothing hits, so you should be able to get rid of this line and the last `NULLIF`.

Comment (Ian2012) [reply tutoroars/templates/oars/jobs/init/oars_init_schemas_tables_users.sh]: Corrected

**Comment by bmtcril** [comment]
@SoryRawyer heads up that this change is coming. I don't think it impacts any of your work right now, but when it lands you'll want to update your test env to match

**Review by bmtcril — APPROVED** [review]
LGTM, I'll merge it tomorrow if no one else comments

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-oars&entry.867055334=https://github.com/openedx/tutor-contrib-oars/pull/55&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-06-08+18:49&entry.1917517419=2023-06-13+17:09&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**